### PR TITLE
feature: Allow removing due dates

### PR DESCRIPTION
**Pull request by YagoGG**

### CODE REVIEW

**Review by theo-damiani — COMMENTED** [review]
Hey @YagoGG ,

Great work, but when tested your PR I run into one problem.
If you create a new to-do list, then you add for example "task1" with a due date
and finally, you add a second task with no due date. If you click on task 1, the update panel is displayed you can remove the due date, add a location. But if you click on the second task, you will see the same due date. 

I tested on your PR #280 and I saw the same problem.
And also if you add a task with a due date, then you remove it, you will see again the due date.

And just to pick one detail, the padding is not the same between the task and the update layout:




But otherwise very good job!

**Comment by theo-damiani** [comment]
> 
> 
> Hey @YagoGG ,
> 
> Great work, but when tested your PR I run into one problem.
> If you create a new to-do list, then you add for example "task1" with a due date
> and finally, you add a second task with no due date. If you click on task 1, the update panel is displayed you can remove the due date, add a location. But if you click on the second task, you will see the same due date.
> 
> I tested on your PR #280 and I saw the same problem.
> And also if you add a task with a due date, then you remove it, you will see again the due date.
> 
> And just to pick one detail, the padding is not the same between the task and the update layout:
> 
> 
> 
> But otherwise very good job!

I just checked the master branch and there is the same problem, so if you want to fix here you can but I will open an issue.

**Review by theo-damiani — APPROVED** [review]

File: app/src/main/java/com/github/steroidteam/todolist/viewmodel/TodoListViewModel.java [hunk 1]
```diff
@@ -81,7 +81,13 @@ public void setTaskDone(int index, boolean isDone) {
 
     public void setTaskDueDate(int index, Date date) {
         Task currentTask = todoListSelected.getValue().getTask(index);
-        todoListRepository.updateTask(selectedTodoList, index, currentTask.setDueDate(date));
+        Task updatedTask;
+        if (date == null) {
+            updatedTask = currentTask.removeDueDate();
```
[the remark this answers is not in the document]
Comment (YagoGG) [reply app/src/main/java/com/github/steroidteam/todolist/viewmodel/TodoListViewModel.java:86]: Keep in mind that is reached when setting a `null` date, which is effectively removing a date.

I agree that the naming is suboptimal, but with the current `Task` API, I feel like this is the best way to approach it. However please do let me know if you have any suggestions on how can it be clarified!

[the remark this answers is not in the document]
Comment (shoutizix) [reply app/src/main/java/com/github/steroidteam/todolist/viewmodel/TodoListViewModel.java:86]: Ohhhh you're right yep that's my bad. So then that's perfect nice !

**Comment by YagoGG** [comment]
Thank you both for the review!

@theo-damiani thanks for the reports, great catches! The padding is going to be fixed with some other visual changes in the dark mode PR, hope that's okay. As for the due date display bug, a fix is available in #313.

I am rebasing with the latest master and merging. Cheers!

**Pull request merged**